libraries/tock-register-interface/README.md

```diff
@@ -3,130 +3,107 @@
-        (0x003 => byte1: ReadWrite<u8>),
-        (0x004 => short: ReadWrite<u16>),
+        // Control register: read-write. `u8` specifies the data type of the
+        // memory-mapped register, and the `Control` parameter specifies how
+        // bitfields within the register should be accessed.
+        0x000 => cr: u8(Control::Register) { Read, Write },
+
+        // Status register: read-only. Registers can have documentation
+        // comments, like:
+        /// Status register
+        0x001 => s: u8(Status::Register) { Read },
+
+        // Registers can have different sizes. The bitfield parameter is
+        // optional:
+        0x002 => byte0: u8 { Read, Write },
+        0x003 => byte1: u8 { Read, Write },
+        0x004 => short: u16 { Read, Write },
 
         // Empty space between registers must be marked with a padding field,
-        // declared as follows. The length of this padding is automatically
-        // computed by the macro.
-        (0x006 => _reserved),
-        (0x008 => word: ReadWrite<u32>),
-
-        // The type for a register can be anything. Conveniently, you can use an
-        // array when there are a bunch of similar registers.
-        (0x00C => array: [ReadWrite<u32>; 4]),
-        (0x01C => ... ),
+        // specified by naming the field _ and omitting the operations list:
+        0x006 => _: u16,
 
-        // Etc.
+        // Padding's length may also be inferred by the macro by specifying _ as
+        // the data type:
+        0x008 => _: _,
 
-        // The end of the struct is marked as follows.
-        (0x100 => @END),
+        // If you have several adjacent registers, you can specify an array
+        // register:
+        0x00C => array: [u32; 4] { Read, Write },
     }
 }
 ```
 
-This generates a C-style struct of the following form.
-
-```rust
-#[repr(C)]
-struct Registers {
-    // Control register: read-write
-    // The 'Control' parameter constrains this register to only use fields from
-    // a certain group (defined below in the bitfields section).
-    cr: ReadWrite<u8, Control::Register>,
-
-    // Status register: read-only
-    s: ReadOnly<u8, Status::Register>
-
-    // Registers can be bytes, halfwords, or words:
-    // Note that the second type parameter can be omitted, meaning that there
-    // are no bitfields defined for these registers.
-    byte0: ReadWrite<u8>,
-    byte1: ReadWrite<u8>,
-    short: ReadWrite<u16>,
+The `peripheral!` macro tests that you specified the fields (registers and
+padding) in increasing order, with no gaps.
 
-    // The padding length was automatically computed as 0x008 - 0x006.
-    _reserved: [u8; 2],
-    word: ReadWrite<u32>,
+The macro generates a trait representing the peripheral with one method for each
+register:
 
-    // Arrays are expanded as-is, like any other type.
-    array: [ReadWrite<u32>; 4],
-
-    // Etc.
+```rust
+trait Registers {
+    // Type aliases (cr, so, byte0) omitted for brevity.
```

Comment:
I don't know what this means. Is "so" a typo?

Reply:
Maybe add a forward reference noting that this will be explained in-depth later?

Reply:
```suggestion
    // Type aliases (cr, s, byte0) omitted for brevity.
```
Would match the running example

Reply:
Yes, it was a type. I corrected it and added the forward reference note.